src/Qowaiv/Text/CharBuffer.cs

```diff
@@ -1,57 +1,112 @@
+                {
+                    buffer[length++] = char.ToUpperInvariant(DiacriticReplace[relace]);
+                }
+                else if (DiacriticLookup.TryGetValue(ch, out var chs))
+                {
+                    buffer[length++] = char.ToUpperInvariant(chs[0]);
+                    buffer[length++] = char.ToUpperInvariant(chs[1]);
+                }
+                else
+                {
+                    buffer[length++] = char.ToUpperInvariant(ch);
+                }
+            }
+        }
+        return length;
     }
 
-    /// <summary>Returns true if the buffer is empty.</summary>
-    [Pure]
-    public bool IsEmpty() => Length == 0;
-
-    /// <summary>Returns true if the buffer is not empty.</summary>
-    [Pure]
-    public bool NotEmpty() => !IsEmpty();
+    [CollectionMutation]
+    public static int ToNonDiacritic(this ReadOnlySpan<char> span, Span<char> buffer, string ignore = "")
+    {
+        var length = 0;
 
-    /// <summary>Gets the first <see cref="char" /> of the buffer.</summary>
-    [Pure]
-    public char First() => buffer[start];
+        foreach (var ch in span)
+        {
+            if (ignore.Contains(ch))
+            {
+                buffer[length++] = ch;
+            }
+            else
+            {
+                var relace = DiacriticSearch.IndexOf(ch);
 
-    /// <summary>Gets the last <see cref="char" /> of the buffer.</summary>
-    [Pure]
-    public char Last() => buffer[end - 1];
+                if (relace != NotFound)
+                {
+                    buffer[length++] = DiacriticReplace[relace];
+                }
+                else if (DiacriticLookup.TryGetValue(ch, out var chs))
+                {
+                    buffer[length++] = chs[0];
+                    buffer[length++] = chs[1];
+                }
+                else
+                {
+                    buffer[length++] = ch;
+                }
+            }
+        }
+        return length;
+    }
 
-    /// <inheritdoc />
-    [Pure]
-    public IEnumerator<char> GetEnumerator() => buffer.Skip(start).Take(Length).GetEnumerator();
+    private const string DiacriticSearch = /*..*/ "ÀÁÂÃÄÅĀĂĄǍǺàáâãäåāăąǎǻÇĆĈĊČƠçćĉċčơÐĎďđÈÉÊËĒĔĖĘĚèéêëēĕėęěÌÍÎÏĨĪĬĮİǏìíîïıĩīĭįǐĴĵĜĞĠĢĝğġģĤĦĥħĶķĸĹĻĽĿŁĺļľŀłÑŃŅŇŊñńņňŉŋÒÓÔÕÖØŌŎŐǑǾðòóôõöøōŏőǒǿŔŖŘŕŗřŚŜŞŠśŝşšŢŤŦţťŧÙÚÛÜŨŪŬŮŰŲƯǓǕǗǙǛùúûüũūŭůűųưǔǖǘǚǜŴŵÝŶŸýÿŷŹŻŽźżž";
+    private const string DiacriticReplace = /*.*/ "AAAAAAAAAAAaaaaaaaaaaaCCCCCCccccccDDddEEEEEEEEEeeeeeeeeeIIIIIIIIIIiiiiiiiiiiJjGGGGggggHHhhKkkLLLLLlllllNNNNNnnnnnnOOOOOOOOOOOooooooooooooRRRrrrSSSSssssTTTtttUUUUUUUUUUUUUUUUuuuuuuuuuuuuuuuuWwYYYyyyZZZzzz";
 
-    /// <inheritdoc />
-    [Pure]
-    [ExcludeFromCodeCoverage]
-    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+    private static readonly Dictionary<char, string> DiacriticLookup = new()
+    {
+        { 'Æ', "AE" },
+        { 'Ǽ', "AE" },
+        { 'æ', "ae" },
+        { 'ǽ', "ae" },
+        { 'ß', "sz" },
```

Comment:
Shouldn't this be `ss` rather than `sz`?

Reply:
Not according to [Wikipedia](https://en.wikipedia.org/wiki/%C3%9F)